ignite/contrib/handlers/time_profilers.py

```diff
@@ -0,0 +1,302 @@
+        e = engine.state.epoch - 1
+        self.event_handlers_times[Events.EPOCH_STARTED][e] = t
+
+        self._dataflow_timer.reset()
+
+    def _as_last_iter_started(self, engine):
+        t = self._event_handlers_timer.value()
+        i = engine.state.iteration - 1
+        self.event_handlers_times[Events.ITERATION_STARTED][i] = t
+
+        self._processing_timer.reset()
+
+    def _as_last_iter_completed(self, engine):
+        t = self._event_handlers_timer.value()
+        i = engine.state.iteration - 1
+        self.event_handlers_times[Events.ITERATION_COMPLETED][i] = t
+
+        self._dataflow_timer.reset()
+
+    def _as_last_epoch_completed(self, engine):
+        t = self._event_handlers_timer.value()
+        e = engine.state.epoch - 1
+        self.event_handlers_times[Events.EPOCH_COMPLETED][e] = t
+
+    def _as_last_completed(self, engine):
+        self.event_handlers_times[Events.COMPLETED][0] = self._event_handlers_timer.value()
+
+        # Remove added handlers:
+        engine.remove_event_handler(self._as_last_started, Events.STARTED)
+
+        #  - add the first handlers
+        events = [
+            Events.EPOCH_STARTED,
+            Events.EPOCH_COMPLETED,
+            Events.ITERATION_STARTED,
+            Events.ITERATION_COMPLETED,
+            Events.COMPLETED
+        ]
+        fmethods = [
+            self._as_first_epoch_started,
+            self._as_first_epoch_completed,
+            self._as_first_iter_started,
+            self._as_first_iter_completed,
+            self._as_first_completed
+        ]
+        lmethods = [
+            self._as_last_epoch_started,
+            self._as_last_epoch_completed,
+            self._as_last_iter_started,
+            self._as_last_iter_completed,
+            self._as_last_completed
+        ]
+
+        for e, m in zip(events, fmethods):
+            engine.remove_event_handler(m, e)
+
+        for e, m in zip(events, lmethods):
+            engine.remove_event_handler(m, e)
+
+    def attach(self, engine):
+        if not isinstance(engine, Engine):
+            raise TypeError("Argument engine should be ignite.engine.Engine, "
+                            "but given {}".format(type(engine)))
+
+        if not engine.has_event_handler(self._as_first_started):
+            engine._event_handlers[Events.STARTED]\
+                .insert(0, (self._as_first_started, (), {}))
+
+    @staticmethod
+    def _compute_basic_stats(data):
+        return OrderedDict([
+            ('min/index', (torch.min(data).item(), torch.argmin(data).item())),
+            ('max/index', (torch.max(data).item(), torch.argmax(data).item())),
+            ('mean', torch.mean(data).item()),
+            ('std', torch.std(data).item()),
+            ('total', torch.sum(data).item())
+        ])
+
+    def get_results(self):
+        events_to_ignore = [
+            Events.EXCEPTION_RAISED,
+            Events.GET_BATCH_COMPLETED,
+            Events.GET_BATCH_STARTED
```

Comment:
We need to check if it is not more simple to replace current logic for measuring dataflow time using these new events.

Reply:
Hello! Latest commit should fix this.